File: 53-maximum-subarray/maximum-subarray.cs
public class Solution {
    public int MaxSubArray(int[] nums) {
        int n = nums.Length;
        int sum =0;
        int maxsum = int.MinValue;
        for(int i=0;i<n;i++){
            sum += nums[i];
            //maxsum = Max(sum,maxsum);
            if(maxsum < sum){
                maxsum = sum;
            }
            if(sum < 0){
                sum =0;
            }
        }
        return maxsum;
    }
}

File: 53-maximum-subarray/maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int prevsum = 0;
        int currsum = 0;
        int maxsum = INT_MIN;
        for(int i=0;i<nums.size();i++){
            currsum = max(nums[i],prevsum + nums[i]);
            maxsum = max(currsum , maxsum);
            prevsum = currsum;
        }
        return maxsum;
    }
};